Sources/EasyPurchase/FetchProduct.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  FetchProduct.swift
+//  EasyPurchase
+//
+//  Created by Sadat Ahmed on 19.09.2023.
+//
+
+import Foundation
+import StoreKit
+
+// Protocol defining actions that can be performed on an in-app product request
+protocol InAppRequestActions: AnyObject {
+    func start()
+    func cancel()
+}
+
+// Protocol extending InAppRequestActions and adding additional properties
+protocol InAppProductRequest: InAppRequestActions {
+    var isCompleted: Bool { get }
+    var cachedProducts: Products? { get }
+}
+
+class FetchProduct : NSObject, SKProductsRequestDelegate, InAppProductRequest {
+    private var productCompletionHandler: ProductComplitionHandler?
+    private var productRequest: SKProductsRequest?
+
+    // Boolean to track if the request is completed
+    var isCompleted: Bool = false
+
+    // Cached products result
+    var cachedProducts: Products?
+
+    init(productIds: Set<String>, productComplitionHandler: @escaping ProductComplitionHandler) {
+        super.init()
+        self.productCompletionHandler = productComplitionHandler
+        productRequest = SKProductsRequest(productIdentifiers: productIds)
```

Comment:
SKProductsRequest 
it may not help you when you go to write unit testing the code as it's not injected.

Reply:
`var delegate: SKProductsRequestDelegate?
init(productIds: Set<String>, productComplitionHandler: @escaping ProductComplitionHandler, productRequest: InAppRequestActions) {
        super.init()
        self.productCompletionHandler = productComplitionHandler
        self.productRequest = productRequest as? SKProductsRequest
        productRequest.delegate = self
    }`
    @RejaulHasan san, Are you talking about this?